env0/resource_environment.go

```diff
@@ -884,6 +909,11 @@ func getCreatePayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 		return client.EnvironmentCreate{}, diag.Errorf("schema resource data deserialization failed: %v", err)
 	}
 
+	//lint:ignore SA1019 reason: https://github.com/hashicorp/terraform-plugin-sdk/issues/817
+	if val, exists := d.GetOkExists("vcs_pr_comments_enabled"); exists {
```

Comment:
if the schema is set with 'vcs_pr_comments_enabled' that's when drifts are detected.
This will work well for customers. (But not perfect).